File: app/src/main/java/com/androidexamples/app/database/TypeDB.java
package com.androidexamples.app.database;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.util.Log;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by marcosvbras on 22/01/17.
 */

public class TypeDB {

    private int id;
    private String type;

    public TypeDB() {}

    public TypeDB(int id, String type) {
        this.id = id;
        this.type = type;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    /**
     * Database section */

    public static final String TABLE_NAME = "TypeDB";
    public static final String COLUMN_ID = "_id";
    public static final String COLUMN_TYPE = "type";

    public static final String TABLE_CREATE = "CREATE TABLE IF NOT EXISTS " + TABLE_NAME + " ("
            + COLUMN_ID + " INTEGER PRIMARY KEY, "
            + COLUMN_TYPE + " TEXT"
            + ");";

    public static String[] columns = new String[] {COLUMN_ID, COLUMN_TYPE};

    public boolean save(Context context) {
        ContentValues contentValues = new ContentValues();
        contentValues.put(COLUMN_TYPE, type);
        ExampleDatabase database = new ExampleDatabase(context);
        long result = database.getDatabase().insert(TABLE_NAME, null, contentValues);
        database.closeConnection();
        return result == 1;
    }

    public boolean update(Context context) {
        ContentValues contentValues = new ContentValues();
        contentValues.put(COLUMN_TYPE, type);
        ExampleDatabase database = new ExampleDatabase(context);
        long result = database.getDatabase().update(TABLE_NAME, contentValues,
                COLUMN_ID + " = ?", new String[]{String.valueOf(id)});
        database.getDatabase().close();
        database.closeConnection();
        return result == 1;
    }

    public boolean delete(Context context) {
        ExampleDatabase database = new ExampleDatabase(context);
        long result = database.getDatabase().delete(TABLE_NAME, COLUMN_ID + " = ?", new String[]{String.valueOf(id)});
        database.closeConnection();
        return result == 1;
    }

    public static List<TypeDB> list(Context context) {
        List<TypeDB> list = null;
        ExampleDatabase database = new ExampleDatabase(context);
        Cursor cursor = database.getDatabase().query(TABLE_NAME, columns, null, null, null, null, null);

        if(cursor.getCount() > 0) {
            list = new ArrayList<>();

            do {
                cursor.moveToFirst();
                TypeDB type = new TypeDB();
                type.setId(cursor.getInt(0));
                type.setType(cursor.getString(1));
                list.add(type);
            } while(cursor.moveToNext());
        }

        database.closeConnection();
        return list;
    }
}

File: app/src/main/java/com/androidexamples/app/database/TypeDB.kt
package com.androidexamples.app.database

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.util.Log

import java.util.ArrayList

/**
 * Created by marcosvbras on 22/01/17.
 */

class TypeDB {

    var id: Int = 0
    var type: String? = null

    constructor() {}

    constructor(id: Int, type: String) {
        this.id = id
        this.type = type
    }

    fun save(context: Context): Boolean {
        val contentValues = ContentValues()
        contentValues.put(COLUMN_TYPE, type)
        val database = ExampleDatabase(context)
        val result = database.database.insert(TABLE_NAME, null, contentValues)
        database.closeConnection()
        return result == 1L
    }

    fun update(context: Context): Boolean {
        val contentValues = ContentValues()
        contentValues.put(COLUMN_TYPE, type)
        val database = ExampleDatabase(context)
        val result = database.database.update(TABLE_NAME, contentValues,
                "$COLUMN_ID = ?", arrayOf(id.toString())).toLong()
        database.database.close()
        database.closeConnection()
        return result == 1L
    }

    fun delete(context: Context): Boolean {
        val database = ExampleDatabase(context)
        val result = database.database.delete(TABLE_NAME, "$COLUMN_ID = ?", arrayOf(id.toString())).toLong()
        database.closeConnection()
        return result == 1L
    }

    companion object {

        /**
         * Database section  */

        val TABLE_NAME = "TypeDB"
        val COLUMN_ID = "_id"
        val COLUMN_TYPE = "type"

        val TABLE_CREATE = ("CREATE TABLE IF NOT EXISTS " + TABLE_NAME + " ("
                + COLUMN_ID + " INTEGER PRIMARY KEY, "
                + COLUMN_TYPE + " TEXT"
                + ");")

        var columns = arrayOf(COLUMN_ID, COLUMN_TYPE)

        fun list(context: Context): List<TypeDB>? {
            var list: MutableList<TypeDB>? = null
            val database = ExampleDatabase(context)
            val cursor = database.database.query(TABLE_NAME, columns, null, null, null, null, null)

            if (cursor.count > 0) {
                list = ArrayList()

                do {
                    cursor.moveToFirst()
                    val type = TypeDB()
                    type.id = cursor.getInt(0)
                    type.type = cursor.getString(1)
                    list.add(type)
                } while (cursor.moveToNext())
            }

            database.closeConnection()
            return list
        }
    }
}
